sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/SystemEventNames.cs

```diff
@@ -58,30 +58,66 @@ public static class SystemEventNames
         /// </summary>
         public const string AcsChatMessageDeleted = "Microsoft.Communication.ChatMessageDeleted";
 
+        /// <summary>
+        /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
+        /// <see cref="AcsChatMessageDeletedInThreadEventData"/> system event.
+        /// </summary>
+        public const string AcsChatMessageDeletedInThread = "Microsoft.Communication.ChatMessageDeletedInThread";
+
         /// <summary>
         /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
         /// <see cref="AcsChatMessageEditedEventData"/> system event.
         /// </summary>
         public const string AcsChatMessageEdited = "Microsoft.Communication.ChatMessageEdited";
 
+        /// <summary>
+        /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
+        /// <see cref="AcsChatMessageEditedInThreadEventData"/> system event.
+        /// </summary>
+        public const string AcsChatMessageEditedInThread = "Microsoft.Communication.ChatMessageEditedInThread";
+
         /// <summary>
         /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
         /// <see cref="AcsChatMessageReceivedEventData"/> system event.
         /// </summary>
         public const string AcsChatMessageReceived = "Microsoft.Communication.ChatMessageReceived";
 
+        /// <summary>
+        /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
+        /// <see cref="AcsChatMessageReceivedInThreadEventData"/> system event.
+        /// </summary>
+        public const string AcsChatMessageReceivedInThread = "Microsoft.Communication.ChatMessageReceivedInThread";
+
+        /// <summary>
+        /// The value of the Event Type stored in <see cref="EventGridEvent.EventType"/> and <see cref="CloudEvent.Type"/> for the
+        /// <see cref="AcsChatThreadCreatedEventData"/> system event.
+        /// </summary>
+        public const string AcsChatThreadCreated = "Microsoft.Communication.ChatThreadCreated";
```

Comment:
Can we write a test for this? With some fancy reflection .

Reply:
Why reflection? We can have a test that attempts to parse this event.

Reply:
In general, we are looking at automating how these mappings are maintained.

Reply:
Not a test for parsing but that all EventDatas are defined in s_systemEventDeserializers

Reply:
Sure, this would be covered by the automation that we are planning so that this does not have to be manually maintained.